### PR TITLE
Remember tags when previewing a story

### LINKED ISSUE #1264: Previewing a submission removes tags
In at least Safari in iOS 17.3, filling in story details and selecting tags, then tapping Preview generates a preview with no tags, and the tags are removed from the form field.

IMG_0531
IMG_0532

### PR DESCRIPTION
**Pull request by epidemian**
This is a late PR after the issue submitted on #1321.

The changes are described on the commit messages. Basically:

* The first commit makes the story preview action not forget the story tags by making `Story#tags_a` work on preview stories (i.e. unsaved). It fixes #1264.
* The second commit makes the story preview show the submitted tags.

I'm not too keen on the approach taken on the 2nd commit. I wish `Story#tags` would Just Work™ on unsaved stories, but alas, it doesn't. Suggestions are welcomed!

The two commits are independent from each other. I left them this way if you prefer to apply the first one without the questionable approach of the second one :)

### CODE REVIEW

File: app/views/stories/_listdetail.html.erb [hunk 1]
```diff
@@ -48,7 +48,7 @@ class="story <%= story.current_upvoted? ? "upvoted" : "" %>
     <% end %>
     <% if story.can_be_seen_by_user?(@user) %>
       <span class="tags">
-        <% story.tags.each do |tag| %>
+        <% (story.previewing ? story.preview_tags : story.tags).each do |tag| %>
```
Comment (epidemian) [inline app/views/stories/_listdetail.html.erb:51]: Oh, btw, with this change, starting the server also makes the files `app/views/home/index.html.erb`  and `app/views/search/index.html.erb` change to have this same thing changed. IDK why that happens. Is that OK? Should i commit those changes too?

Comment (pushcx) [reply app/views/stories/_listdetail.html.erb:51]: This is `config/initializers/heinous_inline_partial` doing its work. Please do commit those changes so the codebase is in a consistent state with what the server will run with.

Comment (epidemian) [reply app/views/stories/_listdetail.html.erb:51]: LOL, that's a poetic name. I've amended those changes onto their appropriate commit!

Comment (epidemian) [reply app/views/stories/_listdetail.html.erb:51]: BTW, i also get this change on  `app/views/home/index.html.erb` when rendering `/stories/new`:

```diff
diff --git a/app/views/home/index.html.erb b/app/views/home/index.html.erb
index 4193138f..633531b8 100644
--- a/app/views/home/index.html.erb
+++ b/app/views/home/index.html.erb
@@ -183,7 +183,7 @@ class="story <%= story.current_upvoted? ? "upvoted" : "" %>
           <span> | </span>
           <span class="dropdown_parent">
             <input id="archive_<%= story.short_id %>" class="archive_button" type="checkbox">
-            <label for="archive_<%= story.short_id %>">caches</label>
+            <label for="archive_<%= story.short_id %>">archive</label>
             <div class="archive-dropdown">
               <a href="<%= story.archiveorg_url %>">Archive.org</a>
               <a href="<%= story.archivetoday_url %>">Archive.today</a>
```

But it also happens on `master` so i guess there's some inconsistency on these partial inlines. I'm excluding that change from this PR as it's unrelated.

Comment (pushcx) [reply app/views/stories/_listdetail.html.erb:51]: Yes,  e3456c07 missed that update, I'll get it. Living up to its name!

**Comment by pushcx** [comment]
Thanks for taking this on. There's a lot of layers of abstraction at work in this bug! And it's a pretty core site feature, so I would rather bring in a slightly-odd fix than leave it broken for longer.

**Comment by epidemian** [comment]
Glad to be of some help :) 
Thanks for maintaining Lobsters!

**Pull request merged**